src/sync/impl/sync_client.hpp

```diff
@@ -50,9 +56,22 @@ struct SyncClient {
 
         client.run(); // Throws
     }) // Throws
+#if NETWORK_REACHABILITY_AVAILABLE
+    , m_reachability_observer(none, [=](const NetworkReachabilityStatus status) {
+        if (status != NotReachable) {
+            cancel_reconnect_delay();
+        }
+    })
     {
+        if (!m_reachability_observer.start_observing()) {
+            m_logger->error("Failed to setup network reachability observer");
+        }
```

Comment:
The object store coding style typically omits braces around the body of an if statement with only one line. There's a number of places in this PR where you've included them.